Source/SimConnect/Source/SimConnectClient.cpp

```diff
@@ -82,6 +226,153 @@ bool SimConnectClient::CloseConnection()
+		return false;
+	}
+
+	if ( SimConnect_AddToDataDefinition( SimConnectHandle, SimConnectDataDefinitionId::AircraftPosition, "Plane Altitude", "feet" ) != S_OK )
+	{
+		LOG( SimConnectClient, Error, "Failed to add 'Plane Altitude' to 'AircraftPosition' data definition." );
+		return false;
+	}
+
+	return true;
+}
+
+// -----------------------------------------------------------------------------
+
+void SimConnectClient::OnConnectionOpen( const SIMCONNECT_RECV_OPEN& Data )
+{
+	LOG( SimConnectClient, Info, "Connected to %s. Simulator Version: %d.%d.%d.%d. SimConnect Version: %d.%d.%d.%d.",
+		Data.szApplicationName,
+		Data.dwApplicationVersionMajor,
+		Data.dwApplicationVersionMinor,
+		Data.dwApplicationBuildMajor,
+		Data.dwApplicationBuildMinor,
+		Data.dwSimConnectVersionMajor,
+		Data.dwSimConnectVersionMinor,
+		Data.dwSimConnectBuildMajor,
+		Data.dwSimConnectBuildMinor
+	);
+}
+
+// -----------------------------------------------------------------------------
+
+void SimConnectClient::OnSimObjectCreated( const SIMCONNECT_RECV_ASSIGNED_OBJECT_ID& Data )
+{
+	const auto RequestHandle = SimConnect::Handle( Data.dwRequestID );
+
+	auto SimObjectPendingCreation = SimObjectsPendingCreation.find( RequestHandle );
+	if ( SimObjectPendingCreation == SimObjectsPendingCreation.end() )
+	{
+		LOG( SimConnectClient, Error, "Failed to find a SimConnect pending creation with handle %lu.", RequestHandle.GetId() );
+		return;
+	}
+
+	CreatedSimObjects.emplace( RequestHandle, Data.dwObjectID );
+
+	SimObjectPendingCreation->second();
+	SimObjectsPendingCreation.erase( SimObjectPendingCreation );
+}
+
+// -----------------------------------------------------------------------------
+
+void SimConnectClient::OnReceivedSimObjectData( const SIMCONNECT_RECV_SIMOBJECT_DATA& Data )
+{
+	const auto DataRequestHandle = SimConnect::Handle( Data.dwRequestID );
+
+	auto AircraftPositionUpdateRegistration = UserAircraftPositionUpdateRegistry.find( DataRequestHandle );
+	if ( AircraftPositionUpdateRegistration == UserAircraftPositionUpdateRegistry.end() )
+	{
+		LOG( SimConnectClient, Error, "Failed to find a registered aircraft position update function with handle %lu.", DataRequestHandle.GetId() );
+		return;
+	}
+
+	const auto* AircraftPositionData = reinterpret_cast< const AircraftPositionDataDefinition* >( &Data.dwData );
+	if ( !AircraftPositionData )
+	{
+		LOG( SimConnectClient, Error, "Received SimObject data update for object %lu but the data was invalid.", Data.dwObjectID );
+		return;
+	}
+
+	const auto CurrentPosition = Utils::EarthCoordinate::FromGeodetic(
+		AircraftPositionData->Latitude,
+		AircraftPositionData->Longitude,
+		AircraftPositionData->Altitude
+	);
+
+	AircraftPositionUpdateRegistration->second( CurrentPosition );
+}
+
+// -----------------------------------------------------------------------------
+
+void SimConnectClient::DispatchProc( SIMCONNECT_RECV* Data, DWORD DataSizeBytes, void* Context )
```

Comment:
Is this supposed to be `void CALLBACK SimConnectClient::DispatchProc` here in the cpp?  The header has this:

```cpp
static void CALLBACK DispatchProc( SIMCONNECT_RECV* Data, DWORD DataSizeBytes, void* Context );
```

(Or does it not matter, like how you can specify default argument values in a header but then don't need to in the cpp?)

Reply:
Yes it should be in the CPP - https://learn.microsoft.com/en-us/cpp/cpp/stdcall?view=msvc-170. It's only safe to omit it for non-static member functions.